youtube_dl/extractor/glomex.py

```diff
@@ -0,0 +1,273 @@
+            player_url = cls._smuggle_origin_url(player_url, origin_url)
+        return player_url
+
+    @classmethod
+    def _match_integration(cls, url):
+        if '_VALID_URL_RE' not in cls.__dict__:
+            cls._VALID_URL_RE = re.compile(cls._VALID_URL)
+        m = cls._VALID_URL_RE.match(url)
+        assert m
+        return compat_str(m.group('integration'))
+
+    @classmethod
+    def _extract_urls(cls, webpage, origin_url):
+        # https://docs.glomex.com/publisher/video-player-integration/javascript-api/
+        EMBED_RE = r'''(?x)
+        (?:
+            <iframe[^>]+?src=(?P<_q1>%(quot_re)s)
+                (?P<url>(?:https?:)?//player\.glomex\.com/integration/[^/]+/iframe-player\.html\?
+                (?:(?!(?P=_q1)).)+)(?P=_q1)|
+            <(?P<html_tag>glomex-player|div)(?:
+                data-integration-id=(?P<_q2>%(quot_re)s)(?P<integration_html>(?:(?!(?P=_q2)).)+)(?P=_q2)|
+                data-playlist-id=(?P<_q3>%(quot_re)s)(?P<id_html>(?:(?!(?P=_q3)).)+)(?P=_q3)|
+                data-glomex-player=(?P<_q4>%(quot_re)s)(?P<glomex_player>true)(?P=_q4)|
+                [^>]*?
+            )+>|
+            # naive parsing of inline scripts for hard-coded integration parameters
+            <(?P<script_tag>script)[^<]*?>(?:
+                (?P<_stjs1>dataset\.)?integrationId\s*(?(_stjs1)=|:)\s*
+                    (?P<_q5>%(quot_re)s)(?P<integration_js>(?:(?!(?P=_q5)).)+)(?P=_q5)\s*(?(_stjs1);|,)?|
+                (?P<_stjs2>dataset\.)?playlistId\s*(?(_stjs2)=|:)\s*
+                    (?P<_q6>%(quot_re)s)(?P<id_js>(?:(?!(?P=_q6)).)+)(?P=_q6)\s*(?(_stjs2);|,)?|
+                (?:\s|.)*?
+            )+</script>
+        )
+        ''' % {'quot_re': r'[\"\']'}
+        for mobj in re.finditer(EMBED_RE, webpage):
+            url, html_tag, video_id_html, integration_html, glomex_player, \
+                script_tag, video_id_js, integration_js = \
+                mobj.group('url', 'html_tag', 'id_html',
+                           'integration_html', 'glomex_player', 'script_tag',
+                           'id_js', 'integration_js')
+            if url:
+                yield cls._smuggle_origin_url(unescapeHTML(url), origin_url)
+            elif html_tag:
+                if html_tag == "div" and not glomex_player:
+                    continue
+                if not video_id_html or not integration_html:
+                    continue
+                yield cls.build_player_url(video_id_html, integration_html, url)
+            elif script_tag:
+                if not video_id_js or not integration_js:
+                    continue
+                yield cls.build_player_url(video_id_js, integration_js, url)
+
+    def _real_extract(self, url):
+        url, origin_url = self._unsmuggle_origin_url(url)
+        embed_id = self._match_id(url)
+        query = compat_parse_qs(compat_urllib_parse_urlparse(url).query)
+        video_id = query['playlistId'][0]
+        # perhaps redundant
```

Comment:
Do you want the first &playlistId=... or the last? Maybe `query['playlistId'][-1]`? Obvs these are probably the same.

Reply:
You are right about the discrepancy between first and last match.

I thought parsing the URL query string would be more robust, yet it will not even get to this point if the `_VALID_URL` regex fails, as the IE will not match.

I will change it per your last suggestion, thus no `_match_id` call or URL parsing.

Thanks